Add a hotkey for disabling the User Interface

PunkPun opened the pull request:
For cinematic recording

PunkPun committed:
Add disable UI hotkey

PunkPun commented:
The UI remains clickable, but I assume that's alright

PunkPun committed:
Move scrolling outside of the draw call

PunkPun committed:
Disable various ingame UI

PunkPun commented:
> Another feature for the cinematic "mode": disable all voices (including notifications)

anvilvapre reviewed (approved):
Untested.

Would rather call it `ChromeEnabled` or `WidgetsEnabled`. `UI` is a bit confusing.

Isn't it better to only allow this in replay mode?

Would also just disable the widget click events? Shouldn't be too difficult?

What is someone by accident enables the mode? Escape doesn't disable it?

michaeldgg2 commented:
I don't know if there has been any changes since last time I tried it. But it would be nice if the UI would reappear, if you pressed Escape, i.e. when the game is paused.

> Isn't it better to only allow this in replay mode?

I think the goal was to give the player the ability to do this directly in-game.

PunkPun commented:
> Would rather call it ChromeEnabled or WidgetsEnabled. UI is a bit confusing.

Renamed to `Ui.WidgetsVisible`

> Would also just disable the widget click events? Shouldn't be too difficult?

We need to keep `ViewportControllerWidget` working.

> What is someone by accident enables the mode? Escape doesn't disable it?

> I don't know if there has been any changes since last time I tried it. But it would be nice if the UI would reappear, if you pressed Escape, i.e. when the game is paused.

I'm looking at escape menu and it also seems to also have a UI disabling capability. It sets the worldRoot visibility to false and has it's own separate menuRoot for the menu to recieve inputs. I think this is the path we should take with this PR as well as it does work cleanly with inputs, although viewport is specifically tied to world and not menus

PunkPun commented:
Added another hotkey to also hide mouse cursor, as well as now UI becomes visible on opening up escape menu

michaeldgg2 reviewed (approved):
LGTM.

I've been using this for some time to record some videos for OpenE2140 and it works great.
